NYPL-discovery/discovery-front-end
# SCC-2572: drb v4

**What's this do?**
Adds support for DRB API v4:
 - Issue queries over GET with updated flat parameter scheme
 - Handle subtle differences in response

DRB_API_BASE_URL newly configurable via env

Includes back support for v3 semantics triggered by presence of '/v3/' in
configured DRB_API_BASE_URL

Also two small things:
 - Fix missing sinon restore in Bib.test 
 - Change: Parallelize drb query with RC search query. Previously, RC search and aggregations query had to complete before starting the DRB API query. With this change, the DRB API query is initiated simultaneously to the RC search and aggregations query because none of these queries depend on each other. This should speed rendering of search results slightly.

**Why are we doing this? (w/ JIRA link if applicable)**
https://jira.nypl.org/browse/SCC-2572

**Do these changes have automated tests?**
Yes. Tests covering `researchNowUtil` and `ResaerchNow`, which were not previously covered.

**How should this be QAed?**
With app configured to connect to both v3 and v4 DRB API, verify that DRB container shows relevant results for RC search queries

**Dependencies for merging? Releasing to production?**
This can go to production as soon as it has been QA'd. It will default to hitting the current V3 API. Once the DRB team begin deployment of DRB API V4, we should disable our drb integration by removing the `drb-integration` feature flag. After DRB V4 is live, we should configure discovery-front-end to connect to the new DRB V4 API , test the feature via url-enabled-features, and finally reenable `drb-integration` generally.

**Has the application documentation been updated for these changes?**

**Did someone actually run this code to verify it works?**
I did
